rh-tictactoe/src/app/screens/Game/index.js

```diff
@@ -40,17 +34,44 @@ class Game extends React.Component {
         history: history.concat([{ squares }]),
         xIsNext: !this.state.xIsNext,
         status,
-        moves
+        moves,
+        stepNumber: history.length
       });
     }
   };
 
+  jumpTo(step) {
+    let status;
+    if (calculateWinner(this.state.history[step])) {
+      status = 'Winner' + calculateWinner(this.state.history[step]);
+    } else {
+      status = 'Next player: ' + (step % 2 == 0 ? 'X' : 'O');
+    }
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0,
+      status
+    });
+  }
+
+  renderMoves = history => {
+    const moves = history.map((step, move) => {
+      const desc = move ? 'Go to move #' + move : 'Go to game start';
+      return (
+        <li key={move}>
+          <button onClick={() => this.jumpTo(move)}>{desc}</button>
+        </li>
+      );
+    });
+    return moves;
+  }
```

Comment:
i think it's time to separate `Game`'s `index` file into `index` and `layout`. What do you think?